crates/blockifier/src/state/state_wrapper.rs

```diff
@@ -0,0 +1,230 @@
+use std::collections::{HashMap, HashSet};
+
+use starknet_api::core::{ClassHash, CompiledClassHash, ContractAddress, Nonce};
+use starknet_api::hash::StarkFelt;
+use starknet_api::state::StorageKey;
+use starknet_api::StarknetApiError;
+use starknet_core::types::FieldElement;
+
+use crate::execution::contract_class::ContractClass;
+use crate::state::errors::StateError;
+use crate::state::state_api::{State, StateReader, StateResult};
+
+#[cfg(test)]
+#[path = "state_wrapper_test.rs"]
+mod test;
+
+/// DynStateWrapper is a wrapper that works as a TransactionState, but using only the dyn State
+/// trait. So it can be used for fallback mechanism.
```

Comment:
Please explain in the comment the problem this new struct is solving and how it is solving it.